### ISSUE TITLE
Getting compile errors when using sentry-elixir in non-plug apps

### ISSUE THREAD

**Issue by myronmarston** [getsentry/sentry-elixir#88]
We've been using this library just fine in a phoenix app.  Today I took a stab at updating some non-phoenix apps (they do not provide an HTTP interface at all, actually -- they just do backend processing) to use this library with the `use_error_logger` option, and now I'm getting a compile error:

```
== Compilation error on file lib/sentry/plug.ex ==
** (CompileError) lib/sentry/plug.ex:66: Plug.Conn.__struct__/0 is undefined, cannot expand struct Plug.Conn
```

Since my app does not use plug, it is not available, and the `%Plug.Conn{}` in the pattern matching fails since the struct can't be expanded.

I think this can be fixed by changing it to not pattern match using the struct syntax. You could probably do `%{__struct__: Plug.Conn}` to enforce that it is a `Plug.Conn` struct w/o requiring `Plug.Conn` to be available at compile time.

**Comment by mitchellhenke** [comment]
@myronmarston Thanks for catching this. 🙂 

I've pushed a fix and released 1.1.2

**Comment by myronmarston** [comment]
Thanks for the quick fix!

**Comment by timbuchwaldt** [comment]
This issue was reintroduced in 2.0.0:

```
== Compilation error on file lib/sentry/plug.ex ==
** (CompileError) lib/sentry/plug.ex:93: Plug.Conn.__struct__/0 is undefined, cannot expand struct Plug.Conn
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
```

**Comment by mitchellhenke** [comment]
@timbuchwaldt sorry about that.  I made a change in https://github.com/getsentry/sentry-elixir/commit/57862f49a6df2e8810dfe4371d726115ab5e45c1 that should be better about preventing that from coming back again.  I've released `2.0.1` that includes the fix.
